internal/querynodev2/handlers.go

```diff
@@ -313,6 +376,11 @@ func (node *QueryNode) searchChannel(ctx context.Context, req *querypb.SearchReq
 		log.Warn("Query failed, failed to get query shard delegator", zap.Error(ErrGetDelegatorFailed))
 		return nil, ErrGetDelegatorFailed
 	}
+	req, err := node.optimizeSearchParams(ctx, req, sd)
+	if err != nil {
```

Comment:
it's OK to continue to execute the request even the optimization failed, it isn't necessary, prefer to only print a warning

Reply:
Actually it depends.
Most error here was for invalid input, which means the search itself is not executable.
Also some index relies on this "optimization"

Reply:
Yes, we won't have a search parameter like ef, which replies on queryhook to generate.